### PR TITLE
feat(tizen-remote): token persistence

### PR DESCRIPTION
**Pull request by boneskull**
This change allows the token to be persisted to disk. The implementation uses lockfiles to avoid concurrent writes to the token cache across instances/processes/etc.

Location of cache & lockfile is OS-dependent.

* * *

- [ ] This needs more tests (E2E)
- [ ] May need to make the cache path configurable for the driver

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* main
  * **PR #36** <a href="https://app.graphite.dev/github/pr/headspinio/appium-tizen-tv-driver/36" target="_blank">Graphite</a> 
    * **PR #37** <a href="https://app.graphite.dev/github/pr/headspinio/appium-tizen-tv-driver/37" target="_blank">Graphite</a> 
      * **PR #38** <a href="https://app.graphite.dev/github/pr/headspinio/appium-tizen-tv-driver/38" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/headspinio/appium-tizen-tv-driver/38?utm_source=stack-comment).

**Comment by boneskull** [comment]
Re the API here:

The `TizenRemote` constructor accepts a boolean `persistToken` flag, which will read and write a token to a persistent store.  Each unique `name` (also an option) for a client is associated with a token.  

* * *

_If `persistToken` is truthy, to determine a token, this is where the class looks:

1. The `token` option passed to the constructor
2. The `TIZEN_REMOTE_TOKEN` env var
3. The persisted token
4. Ask the server for a token

Once we have a token, it is committed to the persisted store.

* * *

_If `persistToken` is falsy_, to determine a token, this is where the class looks:

1. The `token` option passed to the constructor
2. The `TIZEN_REMOTE_TOKEN` env var
3. Ask the server for a token

No tokens are written to the token store.

* * *

The path to the token store (and the lockfile) cannot currently be user-defined; this is OS-dependent.  See [env-paths](https://www.npmjs.com/package/env-paths) for the locations; currently, the "config" path is always used.

Tokens are not encrypted or otherwise encoded.

File: packages/tizen-remote/lib/index.js [hunk 1]
```diff
@@ -289,12 +352,105 @@ export class TizenRemote extends createdTypedEmitterClass() {
...
+   *
+   * Because the `Conf` constructor performs a blocking write to the fs, we need to
+   * wrap the instantiation in a lock.
+   *
+   * Aditionally, any future writes to the cache must be wrapped in the lock.
+   * @returns {Promise<Conf<Record<string,string>>>}
+   */
+  async #getTokenCache() {
+    /** @type {Conf<Record<string,string>>} */
+    let cache;
```
Comment (jlipps) [inline packages/tizen-remote/lib/index.js:420]: seems like we don't need a local variable, could just be:

```js
if (!this.#tokenCache) {
  this.#tokenCache = ...
}
return this.#tokenCache
```

File: packages/tizen-remote/lib/index.js [hunk 2]
```diff
@@ -626,6 +792,8 @@ export class TizenRemote extends createdTypedEmitterClass() {
       this.#debug(`Requesting new token; waiting ${this.#tokenTimeout / 1000}s...`);
       this.#token = await this.getToken();
       this.#debug('Received token');
+    } else if (this.#token) {
+      await this.writeToken(this.#token);
```
Comment (jlipps) [inline packages/tizen-remote/lib/index.js:796]: we don't want to check whether the user wants to persist the token before writing?

Comment (boneskull) [reply packages/tizen-remote/lib/index.js:796]: I think you're right

File: packages/tizen-remote/lib/index.js (hunk 1 above)
Comment (boneskull) [reply packages/tizen-remote/lib/index.js:420]: yeah, this is just getting carried away with my coding style

**Comment by boneskull** [comment]
I've changed the API such that the "is token persistence enabled" conditional lives in the code that reads/writes the cache, so that calling methods needn't also make this check.

**Pull request merged**